addons/pkg-npm/common/src/main/java/org/commonjava/indy/pkg/npm/content/PackageMetadataGenerator.java

```diff
@@ -256,16 +250,31 @@ private boolean writePackageMetadata( List<StoreResource> firstLevelFiles, Artif
 
         DistTag distTags = new DistTag();
         Map<String, VersionMetadata> versions = new LinkedHashMap<>(  );
-        for ( int i = 0; i < versionPaths.size(); i++ )
+        for ( int i = 0; i < packagePaths.size(); i++ )
```

Comment:
little curious here. Seems this can be replaced by for-in loop as packagePaths is sorted, which can drop the usage of list.get(index).

Reply:
Usage of the index is to get the last one from the list and set couple of attributes from it. But yeah, looks I can check if the item is the last via comparing if the version is equal to the last one in the list.